### DISCUSSION TITLE
How can i see if the OpcUa Client has a "working" connection to the Server?

### DISCUSSION THREAD

**Discussion by InformatikerImAll** [eclipse-milo/milo#1047] [General]
Hey. I am new to OpcUa - Milo. How can i check if the connection i established via client.connect().get(); is still connected in lets say 1 hour later?

**Comment by kevinherron** [comment]
Add a `SessionActivityListener` to the client. You have a good connection/session when `onSessionActive` is called, and you've lost it when `onSessionInactive` is called. The client will automatically reconnect and re-establish a Session in the background, indefinitely - no action is necessary.
